teps/0056-pipelines-in-pipelines.md

```diff
@@ -352,6 +363,448 @@ main-`PipelineRun` to be able to fetch and access the sub-`PipelineRun`'s full s
+As such, we will deprecate and remove the existing [`TaskRunSpecs`][taskrunspecs], which contains runtime configuration
+for `TaskRuns` specifically.
+
+```go
+type PipelineRunSpec struct {
+	PipelineRef         *PipelineRef                `json:"pipelineRef,omitempty"`
+	PipelineSpec        *PipelineSpec               `json:"pipelineSpec,omitempty"`
+	RunSpecs            []PipelineTaskRunSpec       `json:"runSpecs,omitempty"`
+	...
+}
+```
+
+```go
+type PipelineTaskRunSpec struct {
+	PipelineTaskName       string                   `json:"pipelineTaskName,omitempty"`
+	ServiceAccountName     string                   `json:"serviceAccountName,omitempty"`
+	PodTemplate            *PodTemplate             `json:"taskPodTemplate,omitempty"`
+	StepOverrides          []TaskRunStepOverride    `json:"stepOverrides,omitempty"`
+	SidecarOverrides       []TaskRunSidecarOverride `json:"sidecarOverrides,omitempty"`
+    Timeouts               *TimeoutFields           `json:"timeouts,omitempty"`	
+}
+```
+
+The example `Pipeline` above named `clone-scan-notify` can be executed in a `PipelineRun` where the configuration for
+executing the child `PipelineRun` - `security-scans` - is optionally specified at runtime, alongside the configuration
+for executing `TaskRuns` (in this example, `git-clone` is also configured at runtime).
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: clone-scan-notify-pipelinerun
+spec:
+  pipelineRef:
+    name: clone-scan-notify
+  runSpecs:
+    - pipelineTaskName: security-scans
+      timeouts:
+        pipeline: "0h0m60s"
+        tasks: "0h0m40s"
+        finally: "0h0m20s"
+    - pipelineTaskName: git-clone
+      serviceAccountName: sa-for-git-clone
+```
+
+The above approach may appear to make it such that users don't need to know if the `PipelineTask` is executed in a 
+`TaskRun`, `Run` or `PipelineRun`. However, users actually need to know because the different runtime-types support 
+different runtime configurations - see [`TaskRun`][taskrunspec], [`Run`][runspec], and [`PipelineRun`][pipelinerunspec].
+In addition, there are other reasons this approach is rejected:
+- Combining the runtime configuration of all runtime types under one field makes the API opaque - users only 
+encounter failures during validation in execution of the `Pipeline`
+- How would we handle providing runtime configuration for a `TaskRun` created in child `PipelineRun` from a 
+`Pipeline` in a `Pipeline`? Would `RunSpecs` field contain `RunSpecs` field? In the [current proposal](#specification),
+ the decoupling of runtime configuration for the different runtime types would allow for `PipelineRunSpecs` to contain
+ `TaskRunSpecs` to solve for this scenario.
+- The name `runSpecs` may lead a user to think that it's configuration for `Run` types only. It's challenging to find 
+  another name to reference all the runtime types - `TaskRuns`, `Runs` and `PipelineRuns`. 
```

Comment:
@skaegi @pritidesai @abayer - added this alternative as discussed during the API working group today, please take a look, thank you!

Reply:
We are lacking concept of `runtime` for `pipelineTask`.

_I think its mentioned in one of the comments in this PR as well or we discussed somewhere - can not recall now_

`pipelineTask` can be `taskRun` (traditional), `pipelineRun` (this proposal), a list of `taskRun`s (matrix), and `run` (custom task). 

From the run time perspective, its all different types of runtimes for a give pipelineTask. The author is aware of each individual type. 

I like the term `pipelineTaskOverrides` so far (due to lack of any better alternative). The idea is to provide a way to specify overrides for `pipelineTask` in turn it could be any type of `run`.

thoughts?